### PR TITLE
Align and update labels and tags with all images

### PR DESCRIPTION
**Pull request by davidcassany**

### CODE REVIEW

File: .obs/dockerfile/slem-base-iso/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # SPDX-License-Identifier: Apache-2.0
 #!BuildTag: suse/sle-micro-iso/base-%%SLEMICRO_VERSION%%:latest
+#!BuildTag: suse/sle-micro-iso/base-%%SLEMICRO_VERSION%%:%VERSION%
+#!BuildTag: suse/sle-micro-iso/base-%%SLEMICRO_VERSION%%:%VERSION%-%RELEASE%
```
Comment (davidcassany) [inline .obs/dockerfile/slem-base-iso/Dockerfile:4]: Use the same tagging criteria for all images. 3 tags approach:

* One for `latest` (handy for CI and testing in general)
* One including the sources version (this our defined version)
* One including the sources version plus the OBS build ID (unique per build)

File: .obs/dockerfile/slem-base-iso/Dockerfile [hunk 2]
```diff
@@ -25,16 +27,18 @@ COPY --from=builder /output /elemental-iso
...
 ARG IMAGE_REPO=$BUILD_REPO/suse/sle-micro-iso/base-$SLEMICRO_VERSION
+ARG IMAGE_TAG=%VERSION%-%RELEASE%
 
 # Define labels according to https://en.opensuse.org/Building_derived_containers
 # labelprefix=com.rancher.slem
 LABEL org.opencontainers.image.title="SLE Micro base-ISO"
-LABEL org.opencontainers.image.description="Includes the SLE Micro ISO"
-LABEL org.opencontainers.image.version="%VERSION%"
+LABEL org.opencontainers.image.description="Includes the SLE Micro base-ISO"
+LABEL org.opencontainers.image.version="${IMAGE_TAG}"
```
Comment (davidcassany) [inline .obs/dockerfile/slem-base-iso/Dockerfile:36]: Setting the version to the most specific image tag -> so even on re-tags we can easily relate it with some specific build.

File: .obs/dockerfile/slem-base-iso/Dockerfile [hunk 3]
```diff
@@ -25,16 +27,18 @@ COPY --from=builder /output /elemental-iso
...
 LABEL org.opencontainers.image.url="https://github.com/rancher/elemental"
 LABEL org.opencontainers.image.created="%BUILDTIME%"
 LABEL org.opencontainers.image.vendor="SUSE LLC"
-LABEL org.opensuse.reference=$IMAGE_REPO
+LABEL org.opencontainers.image.source="%SOURCEURL%"
+LABEL org.opensuse.reference="${IMAGE_REPO}:${IMAGE_TAG}"
```
Comment (davidcassany) [inline .obs/dockerfile/slem-base-iso/Dockerfile:41]: Include the full image reference

File: .obs/dockerfile/slem-base-os/Dockerfile [hunk 4]
```diff
@@ -69,18 +69,18 @@ RUN echo IMAGE_REPO=\"${IMAGE_REPO}\"              >> /etc/os-release && \
 # labelprefix=com.suse.sle.micro
 LABEL org.opencontainers.image.title="SLE Micro"
 LABEL org.opencontainers.image.description="Image containing SLE Micro - a containerized OS layer for Kubernetes."
-LABEL org.opencontainers.image.version="base-%%SLEMICRO_VERSION%%.%RELEASE%"
+LABEL org.opencontainers.image.version="${IMAGE_TAG}"
```
Comment (davidcassany) [inline .obs/dockerfile/slem-base-os/Dockerfile:72]: Use the most specific tag (including version and build ID). The version comes from the github sources tags.

File: .obs/dockerfile/slem-base-os/Dockerfile [hunk 5]
```diff
@@ -69,18 +69,18 @@ RUN echo IMAGE_REPO=\"${IMAGE_REPO}\"              >> /etc/os-release && \
...
 LABEL org.opencontainers.image.url="https://www.suse.com/products/micro/"
 LABEL org.opencontainers.image.created="%BUILDTIME%"
 LABEL org.opencontainers.image.vendor="SUSE LLC"
 LABEL org.opencontainers.image.source="%SOURCEURL%"
-LABEL org.opensuse.reference="registry.suse.com/suse/sle-micro/%%SLEMICRO_VERSION%%:%RELEASE%"
+LABEL org.opensuse.reference="${IMAGE_REPO}:${IMAGE_TAG}"
 LABEL org.openbuildservice.disturl="%DISTURL%"
-LABEL com.suse.supportlevel="alpha"
+LABEL com.suse.supportlevel="l3"
```
Comment (davidcassany) [inline .obs/dockerfile/slem-base-os/Dockerfile:79]: @kkaempf there was a reason to set this to alpha? Our ISOs are `l3`, hence I assume both, OS and ISOs should be on the same support level.

File: .obs/dockerfile/slem-base-os/Dockerfile [hunk 6]
```diff
@@ -69,18 +69,18 @@ RUN echo IMAGE_REPO=\"${IMAGE_REPO}\"              >> /etc/os-release && \
...
 LABEL com.suse.eula="sle-eula"
 LABEL com.suse.lifecycle-url="https://www.suse.com/lifecycle#suse-linux-enterprise-micro"
 LABEL com.suse.image-type="sle-micro"
-LABEL com.suse.release-stage="unreleased"
+LABEL com.suse.release-stage="released"
```
Comment (davidcassany) [inline .obs/dockerfile/slem-base-os/Dockerfile:83]: I assume it should be released if `l3` support level.

File: .obs/dockerfile/slem-flavored-os/Dockerfile [hunk 7]
```diff
@@ -45,6 +38,15 @@ RUN echo IMAGE_REPO=\"${IMAGE_REPO}\"              >> /etc/os-release && \
     echo TIMESTAMP="`date +'%Y%m%d%H%M%S'`"        >> /etc/os-release && \
     echo GRUB_ENTRY_NAME=\"SLE Micro\" >> /etc/os-release
 
+# Define labels according to https://en.opensuse.org/Building_derived_containers
+# labelprefix=com.suse.sle.micro
+LABEL org.opencontainers.image.version="${IMAGE_TAG}"
+LABEL org.opencontainers.image.created="%BUILDTIME%"
+LABEL org.opencontainers.image.source="%SOURCEURL%"
+LABEL org.opensuse.reference="${IMAGE_REPO}:${IMAGE_TAG}"
+LABEL org.openbuildservice.disturl="%DISTURL%"
```
Comment (davidcassany) [inline .obs/dockerfile/slem-flavored-os/Dockerfile:47]: Moving it down to ensure `IMAGE_TAG` and `IMAGE_REPO` are set

**Comment by davidcassany** [comment]
@kkaempf I think we should have the RT image as a flavor, but IIRC you already tried it and for some reason it did not work out, am I right? Which was the problem? At least would be nice to push the RT image to the repository. Having several images into the source repository but RT only in OBS makes maintenance a burden, it is so easy to forget updating the RT image.

File: .obs/dockerfile/slem-base-os/Dockerfile (hunk 5 above)
Comment (kkaempf) [reply .obs/dockerfile/slem-base-os/Dockerfile:79]: 'lack of knowledge' was probably the only reason 😆

File: .obs/dockerfile/slem-base-os/Dockerfile (hunk 6 above)
Comment (kkaempf) [reply .obs/dockerfile/slem-base-os/Dockerfile:83]: Not sure about this one - maintenance might have a different opinion :wink:

**Review by kkaempf — APPROVED** [review]
:+1:

**Pull request merged**